apps/web/src/lib/wagmi/systems/Checker/Slippage.tsx

```diff
@@ -0,0 +1,42 @@
+'use client'
+
+import { Button, type ButtonProps } from '@sushiswap/ui'
+import { type FC, useState } from 'react'
+import { Percent } from 'sushi/math'
+
+const SLIPPAGE_WARNING_THRESHOLD = new Percent(1, 5) // 20%
+
+type SlippageProps = ButtonProps & {
+  text: string
+  slippageTolerance: Percent
+}
+
+const Slippage: FC<SlippageProps> = ({
+  text,
+  slippageTolerance,
+  children,
+  fullWidth = true,
+  size = 'xl',
+  ...props
+}) => {
+  const [open, setOpen] = useState(
+    () => !slippageTolerance.lessThan(SLIPPAGE_WARNING_THRESHOLD),
+  )
+
```

Comment:
When `slippageTolerance` changes, the `open` state does not change.
We could solve this by having a state for having accepted to swap with high slippage, and have `open` be derived.